File: C++/2630.cpp
#include <iostream>
using namespace std;

int main(){
    int testes,r,g,b,res;
    string tipo;
    cin >> testes;
    for(int i = 0; i < testes; i++){
        cin >> tipo;
        cin >> r >> g >> b;
        if(tipo == "eye"){
            res = (0.30*r) + (0.59*g) + (0.11*b);
            cout << "Caso #" << i+1 << ": " << res << endl;
        }else if(tipo == "mean"){
            res = (r+g+b)/3;
            cout << "Caso #" << i+1 << ": " << res << endl;
        }else if(tipo == "min"){
            if(r <= b && r <= g){
                cout << "Caso #" << i+1 << ": " << r << endl;
            }else if(b <= r && b <= g){
                cout << "Caso #" << i+1 << ": " << b << endl;
            }else{
                cout << "Caso #" << i+1 << ": " << g << endl;
            }
        }else{
            if(r >= b && r >= g){
                cout << "Caso #" << i+1 << ": " << r << endl;
            }else if(b >= r && b >= g){
                cout << "Caso #" << i+1 << ": " << b << endl;
            }else{
                cout << "Caso #" << i+1 << ": " << g << endl;
            }
        }
    }
}

File: Python/2630.py
testes = int(input())
for teste in range(testes):
    tipo = input()
    r, g, b = input().split()
    r = int(r)
    g = int(g)
    b = int(b)
    if tipo == "min":
        print("Caso #{0}: {1}".format(teste+1,int(min(r,g,b))))
        
    if tipo == "max":
        print("Caso #{0}: {1}".format(teste+1,int(max(r,g,b))))
        
    if tipo == "mean":
        print("Caso #{0}: {1}".format(teste+1,int((r+g+b)/3)))

    if tipo == "eye":
        print("Caso #{0}: {1}".format(teste+1,int(0.30*r + 0.59*g + 0.11*b)))
